packages/agent/src/connect-v2.ts

```diff
@@ -0,0 +1,199 @@
+
+    // a deeplink to a web5 compatible wallet. if the wallet scans this link it should receive
+    // a route to its web5 connect provider flow and the params of where to fetch the auth request.
+    const walletURI = new URL('web5://connect/');
+    walletURI.searchParams.set(
+      'nonce',
+      Convert.uint8Array(nonce).toBase64Url()
+    );
+    walletURI.searchParams.set('request_uri', parData.request_uri);
+    walletURI.searchParams.set('client_did', clientDid.uri);
+    walletURI.searchParams.set('code_challenge', codeChallengeb64url);
+
+    // call user's callback so they can send the URI to the wallet as they see fit
+    onUriReady(walletURI.toString());
+
+    // subscribe to receiving a response from the wallet with default TTL
+    const tokenURL = Oidc.buildOidcUrl({
+      baseURL    : connectServerUrl,
+      endpoint   : 'token',
+      tokenParam : request.state,
+    });
+
+    /** ciphertext of {@link HybridAuthResponse} */
+    const authResponse = await pollWithTTL(() => fetch(tokenURL));
+
+    if (authResponse) {
+      const jwe = await authResponse?.text();
+
+      // get the pin from the user and use it as AAD to decrypt
+      const pin = await validatePin();
+      const jwt = await Oidc.decryptAuthResponse(clientDid, jwe, pin);
+      const verifiedAuthResponse = (await Oidc.verifyJwt({
+        jwt,
+      })) as HybridAuthResponse;
+
+      // return the grants for liran to
+      console.log(verifiedAuthResponse);
+    }
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+/**
+ * Initiates the wallet connect process. Used when the app (client) wants to import
+ * a delegated identity DID from a wallet (provider).
+ */
+export type WalletConnectOptions = {
+  /** The URL of the intermediary server which relays messages between the client and provider */
+  connectServerUrl: string;
+
+  /**
+   * The URI of the Provider (wallet) which is used to generate the URI returned in `onUriReady`
+   * e.g. `web5://` or `http://localhost:3000/`.
+   *
+   */
+  walletUri: string;
+
+  /**
+   * The protocols of permissions requested, along with the definition and
+   * permission scopes for each protocol. The key is the protocol URL and
+   * the value is an object with the protocol definition and the permission scopes.
+   */
+  permissionRequests: ConnectPermissionRequests;
+
+  /**
+   * The Web5 API provides a URI to the wallet based on the `walletUri` plus a query params payload valid for 5 minutes.
+   * The link can either be used as a deep link on the same device or a QR code for cross device or both.
+   * The query params are `{ request_uri: string; code_challenge: string; }`
+   * The wallet will use the `request_uri to contact the intermediary server's `authorize` endpoint
+   * and pull down the {@link HybridAuthRequest} and use the `code_challenge` to decrypt it.
+   *
+   * @param uri - The URI returned by the web5 connect API to be passed to a provider.
+   */
+  onUriReady: (uri: string) => void;
```

Comment:
Can we rename to `onRequestUriReady` for extra clarity:
```suggestion
  onRequestUriReady: (uri: string) => void;
```

Reply:
It's not the request uri it's the wallet uri.

Reply:
In fact the request uri is in the payload of the wallet URI.

Reply:
Thanks for correcting me, which happens to be case in point LOL: dumb code readers like me get easily confused. So please qualify the URI. Perhaps `onWalletUriRead`, hopefully I get it more right this time.

Reply:
well there's a slight hitch I already ask the user for something called `walletUri` 

`onUriReady` is where I take the `walletUri` they gave me and add the query params to it. any thoughts on a better approach?

Reply:
Done